RecoLocalCalo/EcalRecAlgos/src/EcalRecHitBuilderKernels.cu

```diff
@@ -0,0 +1,738 @@
+      //
+      RecHitFlags_kUnknown                   // to ease the interface with functions returning flags. 
+    };
+    
+    
+    // status code
+    enum EcalChannelStatusCode_Code {
+      kOk=0,
+      kDAC,
+      kNoLaser,
+      kNoisy,
+      kNNoisy,
+      kNNNoisy,
+      kNNNNoisy,
+      kNNNNNoisy,
+      kFixedG6,
+      kFixedG1,
+      kFixedG0,
+      kNonRespondingIsolated,
+      kDeadVFE,
+      kDeadFE,
+      kNoDataNoTP      
+    };
+    
+    
+    
+    
+    
+    __global__
+    void kernel_create_ecal_rehit(
+      // configuration 
+      int const* ChannelStatusToBeExcluded,
+      uint32_t ChannelStatusToBeExcludedSize,   
+      bool const killDeadChannels,
+      bool const recoverEBIsolatedChannels,
+      bool const recoverEEIsolatedChannels,
+      bool const recoverEBVFE,             
+      bool const recoverEEVFE,             
+      bool const recoverEBFE,             
+      bool const recoverEEFE,              
+      float const EBLaserMIN,
+      float const EELaserMIN,
+      float const EBLaserMAX,
+      float const EELaserMAX,
+      // for flags setting
+      int const* expanded_v_DB_reco_flags,    // FIXME AM: to be checked
+      uint32_t const* expanded_Sizes_v_DB_reco_flags,
+      uint32_t const* expanded_flagbit_v_DB_reco_flags,
+      uint32_t expanded_v_DB_reco_flagsSize,
+      uint32_t flagmask,
+      // conditions
+      float const* adc2gev,
+      float const* intercalib,
+      uint16_t const* status,
+      float const* apdpnrefs,
+      float const* alphas,
+      // input for transparency corrections
+      float const* p1,
+      float const* p2,
+      float const* p3,
+      edm::TimeValue_t const* t1,
+      edm::TimeValue_t const* t2,
+      edm::TimeValue_t const* t3,  
+      // input for linear corrections
+      float const* lp1,
+      float const* lp2,
+      float const* lp3,
+      edm::TimeValue_t const* lt1,
+      edm::TimeValue_t const* lt2,
+      edm::TimeValue_t const* lt3,                    
+      // time, used for time dependent corrections
+      edm::TimeValue_t const event_time,
+      // input
+      uint32_t const* did_eb,
+      uint32_t const* did_ee,
+      ::ecal::reco::StorageScalarType const* amplitude_eb,   // in adc counts  
+      ::ecal::reco::StorageScalarType const* amplitude_ee,   // in adc counts  
+      ::ecal::reco::StorageScalarType const* time_eb,   
+      ::ecal::reco::StorageScalarType const* time_ee,   
+      ::ecal::reco::StorageScalarType const* chi2_eb,   
+      ::ecal::reco::StorageScalarType const* chi2_ee,   
+      uint32_t const* flags_eb,
+      uint32_t const* flags_ee,
+      // output
+      uint32_t *did,
+      ::ecal::reco::StorageScalarType* energy,   // in energy [GeV]  
+      ::ecal::reco::StorageScalarType* time,  
+      ::ecal::reco::StorageScalarType* chi2,  
+      uint32_t* flagBits,
+      uint32_t* extra,
+      // other
+      int const nchannels,
+      uint32_t const offsetForInput,
```

Comment:
if I understood correctly the logic, could you rename this to `nChannelsBarrel` (if I understand what it is), write a simple lamba
```c++
auto isEndcap = [=](int channel){ return channel >= nChannelsBarrel; }
```
and use it instead of `ch >= offsetForInput` in the rest of the body, e.g.
```
int const inputCh = isEndcap(ch) ? ch - nChannelsBarrel : ch;
```
?

Reply:
Sorry, I'm being dumb.

Even simpler: can you just do
```
bool isEndcap = (channel >= nChannelsBarrel);
```
and use it everafter ?

Reply:
Correct. Done and it will come in the next commit.